### ISSUE TITLE
test_remote_storage_upload_queue_retries is flaky

### ISSUE THREAD

**Issue by arssher** [neondatabase/neon#3051]
```
2022-12-09T16:04:38.3224170Z =================================== FAILURES ===================================
2022-12-09T16:04:38.3224482Z ______________ test_remote_storage_upload_queue_retries[local_fs] ______________
2022-12-09T16:04:38.3225083Z [gw0] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
2022-12-09T16:04:38.3225585Z test_runner/regress/test_remote_storage.py:314: in test_remote_storage_upload_queue_retries
2022-12-09T16:04:38.3225982Z     assert not churn_while_failpoints_active_thread.is_alive()
2022-12-09T16:04:38.3226239Z E   assert not True
2022-12-09T16:04:38.3226656Z E    +  where True = <bound method Thread.is_alive of <Thread(Thread-3, started 140135066326784)>>()
2022-12-09T16:04:38.3227232Z E    +    where <bound method Thread.is_alive of <Thread(Thread-3, started 140135066326784)>> = <Thread(Thread-3, started 140135066326784)>.is_alive
```

https://github.com/neondatabase/neon/actions/runs/3658686712/jobs/6184058965

**Comment by yeputons** [comment]
Another one: https://github.com/neondatabase/neon/actions/runs/3792409544/jobs/6448666158

**Comment by jcsp** [comment]
This is currently the 2nd most flaky test.

**Comment by arpad-m** [comment]
Last 10 failures found in grafana for example [this](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6872/8013489918/index.html#testresult/8235938280e0f836/retries) or [this](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6885/8011078488/index.html#testresult/9d3f442e027aec01/retries) all hit the same assertion:

```
test_runner/regress/test_remote_storage.py:290: in test_remote_storage_upload_queue_retries
    assert gc_result["layers_removed"] > 0
E   assert 0 > 0
```

**Comment by jcsp** [comment]
Unfortunately this one is still failing (18 times in last 3 days, since the #6898 PR merged.

**Comment by arpad-m** [comment]
I've checked the logs. It's failing with a different error now ([link](https://neon-github-public-dev.s3.amazonaws.com/reports/main/8112922210/index.html#testresult/9a59fe66fb3f4e2f/retries)):

```
Exception: timed out while waiting for <function test_remote_storage_upload_queue_retries.<locals>.<lambda> at 0x7fc627e74040>
```

![Screenshot_20240301_171032](https://github.com/neondatabase/neon/assets/19521842/d0285585-4762-4479-bef7-324daa443cbb)

For some reason, there was a period of low occurences of flakiness before and after my PR was merged, and then it became flaky again.

**Comment by arpad-m** [comment]
Christian told me about #6960 which has touched the test and might have resolved the flakiness. I'll monitor it in the coming days and see whether it is still being flaky, and if no, close this.

**Comment by arpad-m** [comment]
![Screenshot_20240306_155346](https://github.com/neondatabase/neon/assets/19521842/298c5f1e-4374-4cfd-bfe7-6128a4bdc735)

New screenshot. The two green boxes represent weekends where there is little merge/CI activity. The three red lines represent, in order, the merges of 

* https://github.com/neondatabase/neon/pull/6898
* https://github.com/neondatabase/neon/pull/6960
* https://github.com/neondatabase/neon/pull/6999


I'm not sure why there is only little flakiness on February 23, a workday, but #6968 certainly didn't make the situation *worse*.

One can see that after #6960, there was a sharp increase in flakiness for the test. It's also visible that since merging of #6999, the test has only been flaky a single time, [link](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6992/8145808842/index.html#testresult/234515a565eb96de/retries), which was from a CI run for PR #6992, filed before #6999 was merged but after #6960 (not sure why it doesn't point to the merge commit idk, but the last commit from main in that branch was e34059cd185998d8ae60ba3e2086a7258ec6fdb7).

So I'd say we can close this again as fixed by #6999.
